metrics-jetty9/src/main/java/io/dropwizard/metrics5/jetty9/InstrumentedConnectionFactory.java

```diff
@@ -1,21 +1,36 @@
 package io.dropwizard.metrics5.jetty9;
 
-import io.dropwizard.metrics5.Timer;
+import java.util.List;
+import java.util.Optional;
+
 import org.eclipse.jetty.io.Connection;
 import org.eclipse.jetty.io.EndPoint;
 import org.eclipse.jetty.server.ConnectionFactory;
 import org.eclipse.jetty.server.Connector;
 import org.eclipse.jetty.util.component.ContainerLifeCycle;
 
-import java.util.List;
+import io.dropwizard.metrics5.Counter;
+import io.dropwizard.metrics5.Timer;
 
 public class InstrumentedConnectionFactory extends ContainerLifeCycle implements ConnectionFactory {
     private final ConnectionFactory connectionFactory;
     private final Timer timer;
+    private final Optional<Counter> counterMaybe;
```

Comment:
I think a nullable `Counter` would be a better option. We don't do any transformation with the counter and don't expose it, so the advantages of `Optional` are not clear. There's another point: I believe the JVM can optimize out null checks if a final field is null, but I'm not so sure about optionals.

Reply:
`counterMaybe.ifPresent(Counter::inc);` is quite satisfying compared to null-checks, but I'll update